js/store.js

```diff
@@ -68,7 +155,10 @@ const store = {
     const output = {};
     for (const hostname in this._websites) {
       const website = this._websites[hostname];
-      output[hostname] = this.outputWebsite(hostname, website);
+      // only previously ignored websites will have an 'isVisible' key
```

Comment:
I'm pretty sure the above statement could be made not true. As soon as a third party is known not to have a first party on the whitelist it can have the key.
If we simplify this to only ever be two states we should make the code more reliable.